### DISCUSSION TITLE
Cannot connect app to self hosted backend

### DISCUSSION THREAD

**Discussion by ghost** [TomBursch/kitchenowl#222] [Q&A]
Hi,

first of all thanks for this cool project!

I want to self-host kitchenowl but the android app cannot find my server via URL, only via IP and port.
Here is what I did so far:
1. I took the docker-compose.yml file from this repository and changed all three mapped ports.
2. I configured my reverse proxy (HAproxy, including SSL) for the `front` and the `back` HTTP ports, but I did not configure any proxying for the usgwi port. `front` and `back` also have different URLs in my HAproxy.

This lead to the current situation:
I can reach the web ui via browser and via the `front` URL just fine. I can log in and everything seems to work so far.
If I access the `back` via browser using the IP and port, I get a generic error from kitchenowl saying "Request resource not found", which I interpret as "this URL is not used, but the server is running". I get the same message when I browse to `back` via the URL configured in HAproxy.
When I use the android app to connect to my server, everything seems to work just fine if I use the IP and the `back` HTTP port. If I use the URL instead, I get a generic ":/  Hmmmmm.... the server could not be reached". The URL I provided to the android app looked like `https://kitchenowlback.mydomain.tld`.

Does anyone know how I can make the app reach my server via a URL?

Thank you all very much!

**Comment by TomBursch** [comment]
You don't need to expose the backend to the web, it used to be an option but it's not necessary anymore. In the app you should enter the front url instead of the backurl (the same you would type into your browser). I think that should resolve your issue.

**Reply by ghost** [reply]
Thanks for the super fast response!

Unfortunetly, it does not work. To avoid typos, I copied the URL out of my browser but still get same error.

**Reply by ghost** [reply]
Hi, me again. Is it possible that kitchenowl handles SSL certificates somehow? I switched from a certificate from my private authority to a certificate from Let's Encrypt, and now it works. It's a bit surprising for me because my phone trusts my private authority and it works for other services.

But anyways, it works with the frontend URL, just as you proposed. Thank you!

**Reply by TomBursch** [reply]
Great that you figured it out. Hmm yeah I think some other people had problems with certificates too #187. It's a bit strange, because the flutter documentation says that it uses the device's certificates on Android and iOS https://api.flutter.dev/flutter/dart-io/SecurityContext/defaultContext.html
